apps/src/makerlab/BoardController.js

```diff
@@ -28,207 +30,210 @@ const J5_CONSTANTS = {
   SERVO: 4
 };
 
-var BoardController = module.exports = function () {
-  ChromeSerialPort.extensionId = CHROME_APP_ID;
+export default class BoardController {
+  constructor() {
+    ChromeSerialPort.extensionId = CHROME_APP_ID;
 
-  /** @private {five.Board} */
-  this.board_ = null;
+    /** @private {five.Board} */
+    this.board_ = null;
+    /** @private {Object} */
+    this.prewiredComponents = null;
+  }
 
-  this.prewiredComponents = null;
-};
+  connectAndInitialize(codegen, interpreter) {
+    return this.ensureBoardConnected()
+        .then(this.ensureAppInstalled())
+        .then(this.ensureComponentsInitialized.bind(this))
+        .then(this.installComponentsOnInterpreter.bind(this, codegen, interpreter));
+  }
 
-BoardController.prototype.connectAndInitialize = function (codegen, interpreter) {
-  return this.ensureBoardConnected()
-      .then(this.installComponentsOnInterpreter.bind(this, codegen, interpreter))
-      .catch(function (error) {
-        console.log("Board initialization failed:");
-        console.log(error);
-      });
-};
+  connectWithComponents() {
+    return this.ensureBoardConnected()
+        .then(this.ensureComponentsInitialized.bind(this));
+  }
 
-/**
- * Connects to board if not already connected.
- */
-BoardController.prototype.ensureBoardConnected = function () {
-  return new Promise(function (resolve, reject) {
-    ChromeSerialPort.isInstalled(function (error) {
-      if (error) {
-        reject(error);
-        return;
-      }
+  ensureAppInstalled() {
+    return new Promise((resolve, reject) => {
+      ChromeSerialPort.isInstalled(function (error) {
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
+      });
+    });
+  }
 
+  /**
+   * Connects to board if not already connected.
+   */
+  ensureBoardConnected() {
+    return new Promise(function (resolve, reject) {
```

Comment:
It looks like you lost the `this` binding on this method?  Maybe just change to an arrow function.